### PR TITLE
Add supervised_module.py with supervised ML algorithms and planned real-time visualizations

### PR DESCRIPTION
**Pull request by GauravSingh0248**
This PR adds the initial structure for supervised_module.py with core supervised algorithms implemented and placeholders for visualizations.
## 📌 Summary
This PR adds the initial structure for `supervised_module.py` with core supervised algorithms implemented and placeholders for visualizations.

A comprehensive Python module that implements popular supervised learning algorithms with **real-time step-by-step training visualizations** — perfect for educational purposes, model development, and interactive demonstrations.

---

## ✨ Key Features
- **5 Core Algorithms**: Linear Regression, Logistic Regression, Decision Tree, SVM, KNN
- **Real-time Training Visualization**: Watch models learn step-by-step with live plots
- **Multiple Frontend Options**: Streamlit app, Flask web app, and Python module usage
- **Interactive Decision Boundaries**: Visualize classification boundaries evolving during training
- **Performance Metrics Tracking**: Real-time accuracy, MSE, R², ROC curves
- **Dataset Flexibility**: Works with NumPy arrays, Pandas DataFrames, and CSV files

---

## 🎯 Perfect For
- **Educational**: Machine learning courses and workshops
- **Research**: Algorithm comparison and analysis
- **Development**: Model debugging and hyperparameter tuning
- **Presentations**: Interactive client demonstrations

---

## 🛠 Tech Stack
- **Backend**: Python, Scikit-learn, NumPy, Pandas
- **Visualization**: Plotly, Matplotlib, Seaborn
- **Web Frameworks**: Streamlit, Flask
- **Deployment Ready**: Streamlit Cloud, Heroku, AWS/GCP

---

## 📦 Installation & Usage
```bash
pip install -r requirements.txt
streamlit run app.py   # Interactive web app
python web_app.py      # Flask web app
python demo.py         # Python module demo

### CODE REVIEW

**Comment by GauravSingh0248** [comment]
@manasvi-0 PLEASE CHECKOUT THE UPDATES

**Comment by GauravSingh0248** [comment]
So tell me or guide what I have to do now , I can make changes

**Comment by manasvi-0** [comment]
You have used demo.py webapp.py and many more instead 
juist make one file supervised_module.py and add you supervised module.py in app.py file which i have already merged in the main branch
